### ISSUE TITLE
Allow plugins to override mediatypes registered by oteapi-core

### ISSUE THREAD

**Issue by jesper-friis** [EMMC-ASBL/oteapi-core#208]
Currently there is no semantics in how the strategies in oteapi-core stores information in the session. This is addressed by oteapi-dlite by  storing everything in a semantically well-defined collection accessible from the session. But oteapi-dlite is not able to e.g. register a json parser with mediatype `application/json` since that mediatype is already taken by oteapi-core.

The same is the case for accessService.

Current this issue blocks proper use of oteapi-dlite.

The easiest solution would be to move all current strategies in oteapi-core into a separate plugin.

But it might still be useful for the user to have a way to define the precedence between plugins and strategies in the case when two plugins provides a strategy for the same mediaType/accessService.

**Comment by CasperWA** [comment]
Have you tried to do this?
My idea to implement this was that I'd expect two values to appear for a single media type and then have always choose the one that isn't coming from the core, if this happens. But it might be that two values are not returned for the same entry point registration. So have you tried this?

**Comment by jesper-friis** [comment]
Yes, I tried it and oteapi troughs an exception about conflicting mediatypes. 

Splitting the dataresource datamodel into download and parse may solve my problem, since the new `parseType` would determine which parse strategy to use. 

What happens if a plugin defines a download strategy with the same schema as oteapi-core?

**Comment by jesper-friis** [comment]
This has been solved - closing it

**Comment by CasperWA** [comment]
How has this been solved? The last point was that an exception was thrown, has this been resolved by some PR?

**Comment by CasperWA** [comment]
Wrongly closed.
